### PR TITLE
Add Shortcuts

### PR DESCRIPTION
**Pull request by longuint**
I added the option to use shortcuts on each command using Keycode objects. I've also updated the docs to reflect the changes I've made.

### CODE REVIEW

File: src/client/interface/components/shortcut.tsx
Comment (longuint) [inline src/client/interface/components/shortcut.tsx:1]: Couldn't find an efficient way to get access to the `ClientDispatcher` so this file is being stored in the interface.

File: src/client/interface/components/shortcut.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * @name parseShortcutsArray
+ * @description Registers the keybind events for commands.
+ */
+function registerCommandShortcuts(shortcuts: Shortcut, command: BaseCommand) {
+	const dispatcher = CommanderClient.dispatcher();
+
+	const store = useStore();
+
+	// Loop over each array
+	shortcuts.forEach((shortcut, index) => {
```
Comment (longuint) [inline src/client/interface/components/shortcut.tsx:27]: Refactoring this code might make it more understandable.

File: src/client/interface/components/shortcut.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+export function Shortcuts() {
+	// Prevent react renders from causing multiple command executions.
+	if (connections.size() !== 0) return;
+
+	const registry = useMemo(() => CommanderClient.registry(), []);
+
+	// Update connections when new commands are registered.
+	registry.commandRegistered.Connect((command) => {
```
Comment (longuint) [inline src/client/interface/components/shortcut.tsx:75]: I need to write tests to make sure this works.

**Comment by longuint** [comment]
I took a crack at adding a shortcuts panel in the suggestions menu, couldn't quite figure it out. maybe someone could try it out. I was thinking a list of all shortcuts relating to a command, with modifier keys shown with the highlight colors.

**Comment by paradoxuum** [comment]
Haven't got the time to review this yet, but would you be able to set the formatter back to Biome? Seems to be set to Prettier so it looks like there's more changes than there actually are.

You should be able to revert the change to the formatter for `[typescriptreact]` in `.vscode/settings.json` then run `pnpm biome check --write --unsafe` to apply formatting/linting changes to the project. In terms of actually using Biome as a formatter in vscode, you can install the [extension](https://marketplace.visualstudio.com/items?itemName=biomejs.biome).

File: src/shared/core/decorators.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ export enum MetadataKey {
 	Command = "command",
 	Group = "group",
 	Guard = "guard",
+	Shortcut = "shortcut",
```
Comment (paradoxuum) [inline src/shared/core/decorators.ts:10]: `MetadataKey` is only used for decorators, so this should be removed.
```suggestion
```

**Review by paradoxuum — CHANGES_REQUESTED** [review]
Overall LGTM, just some minor changes.

Going forward, the approach I'd like to take with this would be abstracting the shortcuts functionality outside of the interface and using something like `ContextActionService.BindAction` to handle bindings. This is a good starting point, however.

**Comment by longuint** [comment]
Making some changes based off of your suggestions, changes will be pushed soon.

**Comment by longuint** [comment]
> Overall LGTM, just some minor changes.
> 
> Going forward, the approach I'd like to take with this would be abstracting the shortcuts functionality outside of the interface and using something like `ContextActionService.BindAction` to handle bindings. This is a good starting point, however.

The support for `ContextActionService` will continue to be refactored to be more powerful but for now you can bind commands to these actions.

**Review by longuint — COMMENTED** [review]
Implemented suggestions.

**Comment by paradoxuum** [comment]
Looks good, thanks for the PR!

**Pull request merged**